### PR TITLE
Add yank reason to release history page

### LINKED ISSUE #9837: Public release history and release page do not show yanked reason
**What's the problem this feature will solve?**
For yanked versions with a reason provided, PyPI shows this reason on neither the public release history nor the public release page.

**Describe the solution you'd like**
I don't know where exactly it should be shown but I think it should be shown somewhere *at least* on the release page.

**Additional context**
See version 3.4.8 that *has* a yanked reason but it is not shown on the page:
https://pypi.org/project/Red-DiscordBot/#history
https://pypi.org/project/Red-DiscordBot/3.4.8/

### PR DESCRIPTION
**Pull request by cofiem**
Adds the yank reason to the release history page.

Notes:

- I'm happy to change the text colour to blue instead of red, but I thought the red matches the callout on the release page.
- I've added the existing `@include ellipsis;` to ensure the yank reason is always restricted to one line.
- As mentioned in the issue, the full (possibly multi-line) reason is on the release page for those who want to see it.
- I've included  `Reason this release was yanked:` with the screen reader-only class, because it's obvious looking at the release box, but a screenreader might need some context.

Closes #9837

image

### CODE REVIEW

File: warehouse/templates/packaging/detail.html [hunk 1]
```diff
@@ -357,6 +357,12 @@ <h2 class="page-title split-layout">
                 <p class="release__version-date">
                   {{ humanize(hversion.created) }}
                 </p>
+                {% if hversion.yanked and hversion.yanked_reason %}
+                <p class="release__yanked-reason">
+                  <span class="sr-only">{% trans %}Reason this release was yanked:{% endtrans %}</span>
+                  {{ hversion.yanked_reason }}
+                </p>
+                {% endif %}
```
Comment (woodruffw) [inline warehouse/templates/packaging/detail.html:365]: What do you think about making this a `callout-block callout-block--warning` instead? That'll mirror what other views on PyPI do for a warning block. 

For example, here's the warning callout on user-scoped API token creation:

Screenshot 2024-01-31 at 2 49 38 PM

**Review by woodruffw — COMMENTED** [review]
This approach LGTM overall, thank you @cofiem! I left one thought about presentation.

File: warehouse/templates/packaging/detail.html (hunk 1 above)
Comment (cofiem) [reply warehouse/templates/packaging/detail.html:365]: Thanks for the review! On the release page, the callout is 'danger' / red. Should the yank reason callout here match the release page? https://github.com/pypi/warehouse/pull/15078#pullrequestreview-1787211887

Comment (woodruffw) [reply warehouse/templates/packaging/detail.html:365]: Yeah, makes sense to me!

**Review by miketheman — CHANGES_REQUESTED** [review]
Looking good - awaiting the final change for the callout block styling.

**Comment by cofiem** [comment]
Yanked reason styling updated.
image

**Review by woodruffw — APPROVED** [review]
LGTM, thanks @cofiem!

**Comment by miketheman** [comment]
@cofiem Thanks for working through this! I manually purged one example page to confirm, and it looks good: https://pypi.org/project/Red-DiscordBot/#history

The rest of pages should naturally expire from cache in a day, or as they yank/unyank their HTML will be refreshed.

**Pull request merged**